src/app.js

```diff
@@ -102,6 +102,33 @@ class Engagement {
         }
       });
     }
+
+    if (item.classList[0] === 'checkbox') {
+      const index = e.target.parentElement.innerText;
+      this.storage = JSON.parse(localStorage.getItem('tasks-storage'));
+      for (let i = 0; i < this.storage.length; i++) {
+        if (this.storage[i].description === index) {
+          const inde = this.storage.indexOf(this.storage[i]);
+          const checkOption = document.querySelectorAll('.body-task');
+          checkOption[inde].classList.add('strike');
+          this.storage[i].completed = true;
+          for (let j = 0; j < this.storage.length; j++) {
+            if (inde < j) {
+              this.storage[j].index -= 1;
+            }
+          }
+          localStorage.setItem('tasks-storage', JSON.stringify(this.storage));
+          const clearall = document.querySelector('button[type=button]');
+          clearall.addEventListener('click', () => {
+            this.storage = this.storage.filter((check) => check.completed === false);
+            localStorage.setItem('tasks-storage', JSON.stringify(this.storage));
+            const todoList = document.querySelector('.todo-list');
+            todoList.innerHTML = '';
+            window.location.reload();
+          });
+        }
+      }
+    }
```

Comment:
- [ ] I noticed something. The checkbox is not updating the upon user actions on the second click. On the first click, the object's status gets updated to `true` and the task gets struck out. When I click for the second time to update the status, from `true` to `false` it doesn't update. Please see the demo gif to get some understanding of what I am pointing out here.

Demo |
:-------:|
![a1](https://user-images.githubusercontent.com/39523312/166104773-3e5ddb1b-8a9c-4813-a6df-50f1a29ba810.gif)

Reply:
@mmsesay Yes, that was insightful. Apparently, I didn't check for the uncheck box. I have fixed it now